notes

```diff
@@ -117,16 +118,47 @@ generate_name() {
 }
 
 new_note() {
-    local note_name="$*"
-    if [[ $note_name == "" ]]; then
-         note_name="$(generate_name)"
+    local template_used=false
+    local note_name=""
+
+    if [[ "$1" == "-t" ]]; then
+        template_used=true
+        shift
```

Comment:
We should check the number of args here, or somewhere...

If you do something weird right now like `notes -t abc` then it creates a new note called `abc` from template `abc`, which is definitely odd.

I think the logic should be:

* If you pass just one argument, and it's `-t` then we just fail clearly.
* If you pass `-t <arg>` then we use the argument for the template, and use a quicknote name format.
* If you pass `-t <arg1> <arg2>` then arg1 is the template and arg2 is the note name.

Does that make sense?

Reply:
You're right. I think it should work now

Reply:
> You're right. I think it should work now

Not quite - I'm not sure why, but `notes new x` doesn't work any more! If you run `./test.sh` you'll see lots of them are failing.

Those tests weren't run automatically on PRs until now unfortunately, sorry about that, I've just got that working now I think so that should help catch this in future (although I'm not sure if they'll run for this existing PR without a rebase - don't worry about that for now, it's fine).

Reply:
I completely forgot about running those tests. Sorry about that. I'll fix it this weekend. 
Thanks for your patience.